From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sun, 30 Dec 2018 21:13:43 +0900
Subject: ask-password-api: do not call ask_password_keyring() if keyname ==
 NULL

Fixes #11295.

(cherry picked from commit 1f00998c8739ac6adc2b7623cc1e5a8f67d95d7d)
(cherry picked from commit 095a38313daf043413c863634378c8ea7e5f6a09)
---
 src/shared/ask-password-api.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/ask-password-api.c b/src/shared/ask-password-api.c
index 246e27a..6a51c2e 100644
--- a/src/shared/ask-password-api.c
+++ b/src/shared/ask-password-api.c
@@ -341,7 +341,7 @@ int ask_password_tty(
                         goto finish;
                 }
 
-                if (notify >= 0 && pollfd[POLL_INOTIFY].revents != 0) {
+                if (notify >= 0 && pollfd[POLL_INOTIFY].revents != 0 && keyname) {
                         (void) flush_fd(notify);
 
                         r = ask_password_keyring(keyname, flags, ret);
